AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/AWSDataStorePlugin+DataStoreSubscribeBehavior.swift

```diff
@@ -25,6 +23,11 @@ extension AWSDataStorePlugin: DataStoreSubscribeBehavior {
         return publisher.filter { $0.modelName == modelName }.eraseToAnyPublisher()
     }
 
+    public func observe<M: Model>(for modelType: M.Type) async -> AmplifyAsyncThrowingSequence<MutationEvent> {
+        // There's `AWSDataStoreObserveOpertion` operation that can be removed
+        return Fatal.preconditionFailure("Need to return AmplifyAsyncThrowingSequence")
+    }
```

Comment:
Here, we used to only return the Combine publisher from `publisher(modelType:)` API. In this case, we don't need to create a new operation, but thought it would work with the task adapter and returning `task.inProcess` from there, but that is not the case right now. So we can actually just delete `AWSDataStoreObserveOpertion` if it is not useful and make a provider out of the publisher, and pass the provider to the AmplifyAsyncThrowingSequence to be returned?